### DISCUSSION TITLE
parts after # in the url are sometimes useful

### DISCUSSION THREAD

**Discussion by happy15** [EINDEX/logseq-copilot#41] [General]
Hi there, I would like to discuss use cases like mine: the internal doc platform uses parts after # in the url to route to different docs.

For logseq copilot (which is awesome), it seems the parts after # is ignored when search logseq graph. For example, if I open some doc at sub.xxx.com/doc/#id=yyy, all url in the logseq graph which matches sub.xxx.com/doc/ will display in the plugin popup. They are actually different docs, so the search results in this use cases is not that useful.

Whether this due the design of logseq copilot plugin or logseq search api itself?

**Comment by EINDEX** [comment]
Yes, this is logic built in logseq copilot.

This thing has 2 common use cases:
1. Backend rending and static and SPA which use web history routing. 
2. Single-page applications, which like using hash `#` to routing.

In this case 1, `#` is just being id for title things. but for the second case, they will use `#` to routing.

I am considering solutions for this:
1. Try to understand whether the web page is SPA and using the hash route, then bring hash to url.
2. Do nothing on this, because most website is going to rending at the backend. And most articles are running on backend rending websites.
3. Create a rule base to maintain what website using hashing mode.

**Reply by happy15** [reply]
Solution 3 seems like a viable solution, a whitelist of domains or websites to respect hash routing and let the user decide.

This is not urgent and hopefully it will be in the future releases.

Thanks for the discussion.
